XenAdmin/Controls/XenSearch/GroupingControl.cs

```diff
@@ -403,6 +403,7 @@ void button_MouseUp(object sender, MouseEventArgs e)
         {
             draggedButton = null;
             dragging = false;
+            Setup();

```

Comment:
The problem with this solution is that Setup is called and does things twice (first for MouseMove and then for MouseUp). Also, it's called when not needed (when just clicking on the button without dragging it).